### DISCUSSION TITLE
How do I host my Next JS app with dynamic routing in shared hosting ?

### DISCUSSION THREAD

**Discussion by SaidulBadhon** [vercel/next.js#17190] [Help]
Hi, I want to host my next js static html export app to a shared hosting (hostinger.com). When I host my site there, I can access it fine but when I try to reload the website, it goes back to the home page.  I tried using some .htacces code but none of them r working...
I tried, 
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /subdirectory
RewriteRule ^index\.html$ - [L]
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteCond %{REQUEST_FILENAME} !-l
RewriteRule . /index.html [L]
</IfModule>

and

Options -MultiViews
RewriteEngine On
RewriteCond %{REQUEST_FILENAME} !-f
RewriteRule ^ index.html [QSA,L]

its not working. can anyone please help me fix this issue ?

**Comment by ringuralte** [comment]
@SaidulBadhon did you ever find a solution to this problem? I'm currently using query params and swr to do my data fetching cause dynamic routes/pages won't work when i reload the page too.

**Reply by SaidulBadhon** [reply]
nope. I have never found my solution!.

**Reply by bajpaigaurav** [reply]
Hi @SaidulBadhon - mind sharing how you are hosting your app to hostinger ?
what you did after doing next build && export -> have you transferred .next folder to hostinger or did a build there with any modifications.
Thank

**Reply by fadumoaideed** [reply]
You will need to migrate your website from shared hosting to a VPS hosting service to allow for nodejs compatibility. This will not work with shared hosting as your are using SSR. https://www.hostinger.co.uk/tutorials/migrate-website-from-shared-hosting-to-vps

**Comment by jarvissilva937** [comment]
Hosting next js website is very easy on shared hosting you can use Hostinger shared hosting to host next js here is the complete step by step tutorial: [Host next js on hostinger](https://hostwizly.com/host-next-js-website-on-hostinger)

**Reply by Devkant21** [reply]
Only static files works as per the above documentation. If you try to deploy forms over there it doesn't work. You don't get json response. You would need VPS then only the API's which you have created would work.

**Comment by xnivaxhzne** [comment]
If anyone facing the issue of not being able to access the route without adding `.html` extension when hosted the static files in hostinger, make the [trailingSlash](https://nextjs.org/docs/pages/api-reference/next-config-js/trailingSlash) true in next.config.js file. You can add [.htaccess](https://www.geeksforgeeks.org/how-to-remove-html-extension-from-url-of-a-static-page/) though. But with that you can't access the mid level routes without the extension. But making the mentioned next config file change, you can do the following conversion during the build time which does the job.

  > Change links `/me` -> `/me/` and emit `/me.html` -> `/me/index.html`
